src/Datastore/Operation.php

```diff
@@ -405,7 +405,7 @@ public function runQuery(QueryInterface $query, array $options = [])
                     yield $result;
                 }
 
-                if ($query->canPaginate() && $res['batch']['moreResults'] !== 'NO_MORE_RESULTS') {
+                if ($query->canPaginate() && $res['batch']['moreResults'] === 'NOT_FINISHED') {
```

Comment:
Each page can return 300 results. So far as I have seen in the documentation, this is not configurable.